testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -1087,6 +1087,14 @@ private void assertLineage(String baseQuery, Set<String> inputTables, OutputColu
                         .add("mock.default.table_for_output")
                         .build(),
                 outputColumnMetadata);
+
+        runQueryAndWaitForEvents(format("DELETE FROM mock.default.table_for_output WHERE EXISTS (%s) ", baseQuery), 2);
+        assertLineage(
+                ImmutableSet.<String>builder()
+                        .addAll(inputTables)
+                        // This table shouldn't be a part of the input
+                        .add("mock.default.table_for_output")
```

Comment:
This seems like a broken test (until the fix in the last commit), so it should be folded into that commit, instead.